src/cargo/ops/cargo_test.rs

```diff
@@ -85,7 +85,24 @@ fn run_unit_tests(
     } in compilation.tests.iter()
     {
         let test = unit.target.name().to_string();
-        let exe_display = path.strip_prefix(cwd).unwrap_or(path).display();
+
+        let test_path = unit.target.src_path().path().unwrap();
+        let exe_display = if let TargetKind::Test = unit.target.kind() {
+            format!(
+                "{} ({})",
+                test_path
+                    .strip_prefix(unit.pkg.root())
+                    .unwrap_or(&test_path)
+                    .display(),
```

Comment:
this returns an opaque struct that implements Display, which I think makes it harder to collapse these 2 `format!`'s into 1 unless I `to_string()` this call and the `"unittests"` &has str, let me know what people would prefer

Reply:
Nah this seems fine I think, no need to worry too much about perf here!